var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -169,6 +169,7 @@ class Cp2k(MakefilePackage, CMakePackage, CudaPackage, ROCmPackage):
         depends_on("cray-libsci+openmp", when="^[virtuals=blas] cray-libsci")
 
     with when("smm=libxsmm"):
+        depends_on("libxsmm@:1~header-only")
```

Comment:
maybe drop the repeated `~header-only`, once should be enough, then the following 3 `depends_on` statements can be just about version compat bounds

```suggestion
        depends_on("libxsmm~header-only")
        depends_on("libxsmm@:1")
```